src/halodrops/plotting/quicklooks.py

```diff
@@ -0,0 +1,347 @@
+from halodrops import sonde
+from halodrops.helper import paths
+from halodrops.qc import profile
+
+from importlib import reload
+from gogoesgone import processing as pr
+from gogoesgone import zarr_access as za
+
+reload(pr)
+reload(za)
+
+import xarray as xr
+import numpy as np
+import cartopy as cp
+import cartopy.crs as ccrs
+from cartopy.mpl.gridliner import LONGITUDE_FORMATTER, LATITUDE_FORMATTER
+from cartopy.feature import LAND
+import cartopy.feature as cfeature
+from matplotlib.offsetbox import AnchoredText
+import matplotlib.pyplot as plt
+import matplotlib.ticker as mticker
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import matplotlib.dates as mdates
+from datetime import datetime, date
+import s3fs
+import pandas as pd
+
+
+def convert_time_to_str(time=None, time_format="%Y%m%d %H:%M:%S"):
+
+    """
+    Convert input time into desired string format.
+    """
+
+    # Ensure time is in correct format
+    timestamp = (time - np.datetime64("1970-01-01T00:00:00")) / np.timedelta64(1, "s")
+    datetime_time = datetime.utcfromtimestamp(timestamp)
+    str_time = datetime_time.strftime(time_format)
+
+    return str_time
+
+
+def get_mean_launch_time(ds_flight=None, time_format="%Y%m%d %H:%M:%S"):
+
+    """
+    Compute mean launch time from all sondes in the dataset.
+    """
+
+    mean_time = convert_time_to_str(ds_flight.launch_time.mean().values, time_format)
+
+    return mean_time
+
+
+def get_satellite_data(
+    satellite_time="mean_launch_time",
+    time_format="%Y%m%d %H:%M:%S",
+    ds_flight=None,
+    satellite_name="goes16",
+    channel=13,
+    product="ABI-L2-CMIPF",
+    extent=(-62, -48, 10, 20),
+):
+
+    """
+    Access satellite data for nearest time, map to lon/lat grid, and convert to dataset.
+    By default use the mean launch time from dropsonde dataset.
+    """
+
+    # Get correct time for satellite data
+    if satellite_time == "mean_launch_time":
+        use_time = get_mean_launch_time(ds_flight=ds_flight)
+    else:
+        use_time = convert_time_to_str(time=satellite_time)
+
+    # Get filepath to satellite data at nearest time.
+    flist = za.nearest_time_url(use_time, time_format, channel, product, satellite_name)
+    m = za.get_mapper_from_mzz(flist)
+
+    # Select subset of satellite domain
+    img = pr.Image(m)
+    subset = img.subset_region_from_latlon_extents(extent, unit="degree")
```

Comment:
Similar to how we get mean launch time from the dropsondes, I think the default extents should also be dependent on the actual launch locations, rather than a fixed bounding box.